cmake/modules/FindZephyr-sdk.cmake

```
@@ -60,20 +60,66 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
-                 /usr
-                 /usr/local
-                 /opt
-                 $ENV{HOME}
-                 $ENV{HOME}/.local
-                 $ENV{HOME}/.local/opt
-                 $ENV{HOME}/bin)
+    # Paths that are used to find installed Zephyr SDK versions
+    SET(zephyr_sdk_search_paths
+        /usr
+        /usr/local
+        /opt
+        $ENV{HOME}
+        $ENV{HOME}/.local
+        $ENV{HOME}/.local/opt
+        $ENV{HOME}/bin)
+
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS ${zephyr_sdk_search_paths})
+
+    # Remove duplicate entries and sort naturally in descending order.
+    set(zephyr_sdk_found_versions ${Zephyr-sdk_CONSIDERED_VERSIONS})
+    set(zephyr_sdk_found_configs ${Zephyr-sdk_CONSIDERED_CONFIGS})
+
+    list(REMOVE_DUPLICATES Zephyr-sdk_CONSIDERED_VERSIONS)
+    list(SORT Zephyr-sdk_CONSIDERED_VERSIONS COMPARE NATURAL ORDER DESCENDING)
+
+    # Loop over each found Zepher SDK version until one is found that is compatible.
+    foreach(zephyr_sdk_canidate ${Zephyr-sdk_CONSIDERED_VERSIONS})
```

canidate ?
```suggestion
    foreach(zephyr_sdk_candidate ${Zephyr-sdk_CONSIDERED_VERSIONS})
```
and other locations as well.

---

Fixed